hypertrace-metrics-generator/hypertrace-metrics-generator/src/main/java/org/hypertrace/metrics/generator/MetricsProcessor.java

```diff
@@ -0,0 +1,169 @@
+import static org.hypertrace.metrics.generator.MetricsConstants.STATUS_CODE;
+import static org.hypertrace.metrics.generator.MetricsConstants.TENANT_ID_ATTR;
+import static org.hypertrace.metrics.generator.MetricsGenerator.METRICS_GENERATOR_JOB_CONFIG;
+import static org.hypertrace.metrics.generator.MetricsGenerator.OUTPUT_TOPIC_PRODUCER;
+
+import com.typesafe.config.Config;
+import io.opentelemetry.proto.metrics.v1.ResourceMetrics;
+import java.time.Duration;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+import org.apache.kafka.streams.KeyValue;
+import org.apache.kafka.streams.kstream.Transformer;
+import org.apache.kafka.streams.processor.Cancellable;
+import org.apache.kafka.streams.processor.ProcessorContext;
+import org.apache.kafka.streams.processor.PunctuationType;
+import org.apache.kafka.streams.processor.To;
+import org.apache.kafka.streams.state.KeyValueIterator;
+import org.apache.kafka.streams.state.KeyValueStore;
+import org.hypertrace.metrics.generator.api.v1.Metric;
+import org.hypertrace.metrics.generator.api.v1.MetricIdentity;
+import org.hypertrace.viewgenerator.api.RawServiceView;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class MetricsProcessor
+    implements Transformer<String, RawServiceView, KeyValue<byte[], ResourceMetrics>> {
+
+  private static final Logger logger = LoggerFactory.getLogger(MetricsProcessor.class);
+
+  private static final String DELIMITER = ":";
+  private static final String METRIC_AGGREGATION_TIME_MS = "metric.aggregation.timeMs";
+  private static final String METRIC_EMIT_WAIT_TIME_MS = "metric.emit.waitTimeMs";
+
+  private ProcessorContext context;
+  private KeyValueStore<MetricIdentity, Long> metricsIdentityStore;
+  private KeyValueStore<MetricIdentity, Metric> metricsStore;
+  private long metricAggregationTimeMs;
+  private long metricEmitWaitTimeMs;
+  private To outputTopicProducer;
+
+  @Override
+  public void init(ProcessorContext context) {
+
+    this.context = context;
+    this.metricsIdentityStore =
+        (KeyValueStore<MetricIdentity, Long>)
+            context.getStateStore(MetricsGenerator.METRICS_IDENTITY_STORE);
+    this.metricsStore =
+        (KeyValueStore<MetricIdentity, Metric>)
+            context.getStateStore(MetricsGenerator.METRICS_IDENTITY_VALUE_STORE);
+    this.outputTopicProducer = To.child(OUTPUT_TOPIC_PRODUCER);
+
+    Config jobConfig = (Config) (context.appConfigs().get(METRICS_GENERATOR_JOB_CONFIG));
+    this.metricAggregationTimeMs = jobConfig.getLong(METRIC_AGGREGATION_TIME_MS);
+    this.metricEmitWaitTimeMs = jobConfig.getLong(METRIC_EMIT_WAIT_TIME_MS);
+
+    restorePunctuators();
+  }
+
+  @Override
+  public KeyValue<byte[], ResourceMetrics> transform(String key, RawServiceView value) {
+    // construct metric attributes & metric
+    Map<String, String> attributes = new HashMap<>();
```

Comment:
As discussed we are doing application level aggregation versus using inbuilt aggregation by prometheus, since prometheus polls metrics at fixed duration instead of realtime consumption